feat: Update README.md

mvlassis opened the pull request:
Closes #5 

This PR updates the `README.md` and the description on `metadata.yaml` for the charm.

For the Charmhub home page Markdown, I have created a [Discourse post](https://discourse.charmhub.io/t/18854) to host the page, and linked to that page in `metadata.yaml`.

afgambin commented on metadata.yaml:
```diff
@@ -1,9 +1,10 @@
 
 description: |
-  Charm for providing access to Azure service principal credentials
+  azure-auth-integrator is an integrator charm responsible for relaying the credentials required to be able to interact with Microsoft Entra ID using service principals.
```
```suggestion
  azure-auth-integrator is an integrator charm responsible for relaying the credentials required to interact with Microsoft Entra ID using service principals.
```

afgambin commented on README.md:
```diff
@@ -1,2 +1,51 @@
+`azure-auth-integrator` is an open-source project that welcomes community contributions, suggestions,
+fixes and constructive feedback.
+
+- Report [issues](https://github.com/canonical/azure-auth-integrator/issues)
```
```suggestion
- Report [issues](https://github.com/canonical/azure-auth-integrator/issues).
```

afgambin commented on README.md:
```diff
@@ -1,2 +1,51 @@
+
+- Report [issues](https://github.com/canonical/azure-auth-integrator/issues)
+- [Contact us on Matrix](https://matrix.to/#/#charmhub-data-platform:ubuntu.com)
+- Explore [Canonical Data & AI solutions](https://canonical.com/data)
```
```suggestion
- Explore [Canonical Data & AI solutions](https://canonical.com/data).
```

afgambin commented on README.md:
````diff
@@ -1,2 +1,51 @@
+juju deploy azure-auth-integrator --channel 1/edge
+```
+
+Then, configure the charm with your Azure credentials:
````
```suggestion
Now configure it with your Azure credentials:
```

afgambin reviewed (changes requested):
Nice job @mvlassis, left you some suggestions (I didn't realize and also updated the Discourse post directly - you'll have to double check it). Ty!

mvlassis commented:
@afgambin  Addressed all comments and updated the Discourse post at https://discourse.charmhub.io/t/charmhub-home-page-for-azure-auth-integrator/18854

afgambin reviewed (approved):
LGTM, nice job :)
